Src/IronPython/Runtime/Bytes.cs

```diff
@@ -48,8 +52,16 @@ public class Bytes : IList<byte>, IReadOnlyList<byte>, IEquatable<Bytes>, ICodeF
             _bytes = new byte[size];
         }
 
-        private Bytes(byte[] bytes) {
-            _bytes = bytes;
+        public Bytes([NotNull]byte[] bytes) {
```

Comment:
Yes, they are, but `ReadOnlyMemory` defines [implicit conversions from those types](https://docs.microsoft.com/en-us/dotnet/api/system.readonlymemory-1.op_implicit?view=netcore-3.1). So we do need these overloads to break the abiguity between `IList<byte>` and `ReadOnlyMemory<byte>` when calling from C#. Try to delete the `byte[]` overload and compile the project to see what I mean.

Reply:
Ugh, that is super unfortunate. It means that any time we have an `IList<byte>` and `ReadOnlyMemory<byte>` overload we need to add overloads for `byte[]`, `ArraySegment<byte>` and whatever other type implements `IList<byte>` with an implicit `ReadOnlyMemory<byte>` cast?

Reply:
Well, not really, we don't need to. 

First of all, this affects only calls from C#. For the calls from Python, the overload resolver seems not to take implicit casts into consideration. That's why currently a call with a `Memory` argument fails to resolve, rather than being bound to the `ReadOnlyMemory` overload. Which is a good thing in this case, because it means than in the absence of the `byte[]` overload, a `byte[]` argument call will simply bind to the `IList<byte>` overload, rather than trigger the ambiguous match error. I wouldn't even know how to explicitly "cast" an argument to a specific type in a dynamic language. 

So those overloads are there only for the sake of C# calls. Without them, every call with, say, `byte[] arr`, would either have to use `((IList<byte>)arr)` or `arr.ToMemory()` instead of simply `arr` as argument. But it is till workable. However, in this particular case, I thought that `Bytes` is such a fundamental Python type, basically a Python equivalent of `byte[]`, and initializing it with `byte[]` is such a basic operation that an explicit overload of `Bytes` constructor is warranted. 

I feel much less strongly about the `ArraySegment` overload. It is there just for completeness. I know of no other .NET library type than these two that implement `IList<byte>` and at the same time have an implicit conversion to `ReadOnlyMemory<byte>`.  The IronPython codebase does not use `ArraySegment`, so this overload is there just as courtesy for user C# code. If you think it goes too far, I am ready to remove it. There is always some benefit of having fewer overloads to consider for the overload resolver.

Reply:
There is yet another possibility: remove the two new overloads **together** with the `IList<byte>` and `IEnumerable<object?>` overloads. Then use type tests in the `object?` overload to perform the most efficient conversion to array.

Reply:
Yeah, I though of that, although it's not very .NET-like. I'd prefer to reserve the object overload for Python objects.

Anyway, we can keep the `byte[]` overload if you think it's a good idea.

I would also be fine with having the ambiguous call issue. Just looking at where it fails, in `_bytesio.cs` is shouldn't even be making a copy, the underlying buffer should be exposed directly. I guess the other failures in the `IronPython.SQLite` project could just be solved via a cast.